src/services/services.ts

```diff
@@ -3897,6 +3911,9 @@ module ts {
 
             let typeChecker = program.getTypeChecker();
             let symbol = typeChecker.getSymbolAtLocation(node);
+            if(!symbol && isDeclarationKeyword(node)) {

```

Comment:
as i mentioned earlier, i would check for function keyword here and special case it.